network/lwip2transport/udp_test.go

```diff
@@ -35,15 +36,43 @@ func TestUDPResponseWriterCloseNoDeadlock(t *testing.T) {
 	err := h.ReceiveTo(&noopLwIPUDPConn{localAddr}, []byte{}, destAddr)
 	require.NoError(t, err)
 
-	// Close this single session (i.e. the request sender), it will close proxy.respWriter
-	// udpHandler must make sure only one `Close()` is called, and there should be no deadlocks
-	err = proxy.respWriter.Close()
-	require.NoError(t, err)
-	require.Exactly(t, 1, proxy.closeCnt)
+	// Close proxy.respWriter, it should indirectly Close the proxy only once.
+	const ConcurrentCloseCount = 50
+	errs := make([]error, ConcurrentCloseCount)
+	wg := sync.WaitGroup{}
+	wg.Add(ConcurrentCloseCount)
+	for i := 0; i < ConcurrentCloseCount; i++ {
+		go func(k int) {
+			defer wg.Done()
+			errs[k] = proxy.respWriter.Close()
+		}(i)
+	}
+	wg.Wait()
+
+	nNilErr := 0
+	for _, e := range errs {
```

Comment:
Perhaps a simpler implementation is to create an error channel that you push the errors, then here you ensure the first is nil, while the others are closed in the for loop